homeassistant/components/bluetooth_le_tracker/device_tracker.py

```diff
@@ -19,6 +22,17 @@ def setup_scanner(hass, config, see, discovery_info=None):
     # pylint: disable=import-error
     import pygatt
     new_devices = {}
+    hass.data.setdefault(DATA_BLE, {DATA_BLE_ADAPTER: None})
+
+    async def async_stop(event):
+        """Try to shut down the bluetooth child process nicely."""
+        # These should never be unset at the point this runs, but just for
+        # safety's sake, use `get`.
+        adapter = hass.data.get(DATA_BLE, {}).get(DATA_BLE_ADAPTER)
+        if adapter is not None:
+            adapter.kill()
```

Comment:
This doesn't seem async safe as it contains blocking calls. We're not allowed to block the event loop. Coroutines are executed in the event loop.

Reply:
Fix in #23225